packages/metal-dom/src/dom.js

```diff
@@ -306,28 +306,32 @@ class dom {
 	 * @protected
 	 */
 	static handleDelegateEvent_(event) {
-		dom.normalizeDelegateEvent_(event);
 		var currElement = core.isDef(event[NEXT_TARGET]) ?
-			event[NEXT_TARGET] :
-			event.target;
+				event[NEXT_TARGET] :
+				event.target;
 		var ret = true;
-		var container = event.currentTarget;
-		var limit = event.currentTarget.parentNode;
-		var defFns = [];
 
-		while (currElement && currElement !== limit && !event.stopped) {
-			event.delegateTarget = currElement;
-			ret &= dom.triggerMatchedListeners_(container, currElement, event, defFns);
-			currElement = currElement.parentNode;
-		}
+		if (dom.isAbleToInteractWith_(currElement, event.type)) {
```

Comment:
You're right, this is getting more complex. I think that the complexity is coming mainly from the for/while loops though, not just this new if case, so ideally we should instead find a way to break this method into smaller parts. There could be a function that covers to logic of running the default listeners for example, a `triggerDefaultFns_(event, defFns)`, and another for the while loop in the same way. This would clean this up a lot, don't you think?

If you wish you could do this in this pr, or it could be a separate pr sent later, so that we can merge this fix sooner. For this fix to get in, the only thing missing right now is a test :)